.github/workflows/ci.yml

```diff
@@ -1,6 +1,12 @@
 name: CI
 
-on: [ push, pull_request ]
+on:
+  push:
+    branches:
+      - main
+  pull_request:
+    branches:
+      - main
```

Comment:
Are you sure that the filter `branch` is about the target branch? or is that the actual branch name? in the latter case, main won't match any branch for PRs

Reply:
Yes,

You can see it with the current PR that triggers the pull_request CI.
And directly from the doc: https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#example-including-branches
> For example, the following workflow would run whenever there is a pull_request event for a pull request targeting:
    A branch named main (refs/heads/main)
    A branch named mona/octocat (refs/heads/mona/octocat)
    A branch whose name starts with releases/, like releases/10 (refs/heads/releases/10)
>
```yaml
on:
  pull_request:
    # Sequence of patterns matched against refs/heads
    branches:
      - main
      - 'mona/octocat'
      - 'releases/**'
```

Reply:
thx